### DISCUSSION TITLE
How to avoid listening trait/property changes on undo action for components

### DISCUSSION THREAD

**Discussion by FaisalShaikhHA** [GrapesJS/grapesjs#4969] [Q&A] [answered]
- I have added a custom columns component with a number trait name columns used for adding or removing the child component column.
- When Grapesjs undo remove change it appends the removed column and reverts the column trait as well, but this column trait change triggers this.listenTo(this, "change:columns", this.updateColumns); which adds the new column as well. If this is the case then how to avoid listening to the column trait change on undo or when undoing the removed change Grapesjs should append the removed element first and then revert all other property/trait changes (undo in reverse order) or any other way to handle this situation?
- [Github Issue](https://github.com/GrapesJS/grapesjs/issues/4947) has more detail and a reproducible demo.

**Comment by artf** [comment, marked as answer]
@FaisalShaikhHA if you have already a bug issue opened, there is no need to create also a discussion
